tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -765,6 +762,61 @@ Status StorageManagerCanonical::delete_fragments(
   return Status::Ok();
 }
 
+void StorageManagerCanonical::delete_group(const char* group_name) {
+  Status st;
+  if (group_name == nullptr) {
+    throw Status_StorageManagerError(
+        "[delete_group] Group name cannot be null");
+  }
+
+  auto group_dir = GroupDirectory(
+      vfs_,
+      compute_tp_,
+      URI(group_name),
+      0,
+      std::numeric_limits<uint64_t>::max());
+
+  auto group_detail_uris = group_dir.group_detail_uris();
+  auto group_meta_uris = group_dir.group_meta_uris();
+  auto group_meta_uris_to_vacuum = group_dir.group_meta_uris_to_vacuum();
+  auto group_meta_vac_uris_to_vacuum =
+      group_dir.group_meta_vac_uris_to_vacuum();
+  auto group_file_uris = group_dir.group_file_uris();
+
+  // Delete the group detail files
+  throw_if_not_ok(
```

Comment:
We could make a small helper function for this... removefiles(std::vector<URI> ...)